src/dispatch/case/flows.py

```diff
@@ -393,6 +447,31 @@ def case_status_transition_flow_dispatcher(
 ):
     """Runs the correct flows based on the current and previous status of the case."""
     match (previous_status, current_status):
+        case (CaseStatus.closed, CaseStatus.new):
```

Comment:
Suggest moving these to the bottom of the function. It naturally progress cleanly towards the ending status of `Closed` then.

Reply:
Reordered the cases in order of `current_status`